R/root.R

```diff
@@ -1,87 +1,233 @@
+      ## We may need to relax this, but it's not really clear to me
+      ## how the user gets into this position; they have a bunch of
+      ## files there and they want to a root into it?
+      ##
+      ## One option is provide a boolean arg to proceed anyway in this
+      ## case, at the moment there's not a lot that can be done to
+      ## undo this situation.
       if (length(dir(path, all.files = TRUE, no.. = TRUE)) > 0) {
-        stop("'path' exists but is not empty, or an outpack archive")
+        cli::cli_abort(c(
+          "'path' exists but is not empty, or an outpack archive",
+          i = "Please have a chat with us if this is something you need to do"))
       }
     }
   }
 
-  if (file.exists(file.path(path, ".outpack"))) {
-    root <- outpack_root_open(path, FALSE)
+  config <- config_new(path_archive, use_file_store, require_complete_tree,
+                       logging_console, logging_threshold)
+
+  path_outpack <- file.path(path, ".outpack")
+  if (file.exists(path_outpack)) {
+    root <- root_open(path, locate = FALSE, require_orderly = FALSE)
+    root_validate_same_configuration(match.call(), config, root, environment())
   } else {
-    root <- outpack_init(path, ...)
+    fs::dir_create(path_outpack)
+    fs::dir_create(file.path(path_outpack, "metadata"))
+    fs::dir_create(file.path(path_outpack, "location"))
+    config_write(config, path)
+    root <- outpack_root$new(path)
+    outpack_log_info(root, "init", path, "orderly2::orderly_init")
+  }
+
+  if (!has_orderly_config) {
+    writeLines(empty_config_contents(), file.path(path, "orderly_config.yml"))
   }
-  writeLines(empty_config_contents(), file.path(path, "orderly_config.yml"))
-  invisible(orderly_root(root, locate = FALSE))
+
+  root <- root_open(path, locate = FALSE, require_orderly = TRUE,
+                    call = environment())
+  invisible(root$path)
 }
 
 
-## In orderly, there's quite a bit more to read here, coping with the
-## configuration (orderly_config.yml) which also marks the root.
-##
-## In orderly, the key fields in the configuration are:
-##
-## Things that will be removed
-##
-## * destination - orderly's internal db, now ignored
-## * tags - removed as unused
-##
-## Things that will be entirely reworked
-##
-## * remote - configuration of remotes, will get a full change as this
-##   has lots of issues as is, and because this interacts with
-##   outpack's 'location' support
-## * changelog - will get overhauled because it's not clear that it works
-##   well at the moment
+empty_config_contents <- function() {
+  'minimum_orderly_version: "1.99.0"'
+}
+
+
+## There's quite a few scenarios here:
 ##
-## Things that will probably come in without too much change:
+## * find an existing outpack root that is not an orderly root
+##   - sometimes require that we upgrade (e.g., run) [require orderly t]
+##   - sometimes just work with it (e.g., search, extract, copy) [r o f]
```

Comment:
What's `require orderly t`? And `r o f`?

Reply:
wtf

Reply:
so, I think this was "require orderly = TRUE" and "require orderly = FALSE"